spark2/src/test/java/org/apache/iceberg/spark/source/TestSparkTableUtil.java

```diff
@@ -369,6 +376,84 @@ public void testImportUnpartitionedWithWhitespace() throws Exception {
     }
   }
 
+  public static class GetPartitions {
+
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    // This logic does not really depend on format
+    private final FileFormat format = FileFormat.PARQUET;
+
+    @Test
+    public void testPartitionScan() throws Exception {
+
+      List<ThreeColumnRecord> records = Lists.newArrayList(
+          new ThreeColumnRecord(1, "ab", "data"),
+          new ThreeColumnRecord(2, "b c", "data"),
+          new ThreeColumnRecord(1, "b c", "data"),
+          new ThreeColumnRecord(2, "ab", "data"));
+
+      File location = temp.newFolder("partitioned_table");
+      String tableName = "external_table";
+
+      spark.createDataFrame(records, ThreeColumnRecord.class)
+          .write().mode("overwrite").format(format.toString())
+          .partitionBy("c1", "c2").saveAsTable(tableName);
+
+      TableIdentifier source = spark.sessionState().sqlParser()
+          .parseTableIdentifier(tableName);
+
+      Map<String, String> partition1 = Stream.of(
```

Comment:
I suppose we cannot use Java9 factory methods in Spark2 project

Reply:
:( I think we still are running against java 8

Reply:
That said if you are looking for a better method here we do have all of guava, so you can use
Maps.of( k1, v1, k2,v2, ....)

Or 
ImmutableMaps.of(....)

Or

ImmutableMaps.Builder

Reply:
Oh right forgot we have guava, thanks, changed, its much cleaner now.